### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Tetre2**

### CODE REVIEW

File: src/main/java/ProjectOcean/IO/StudyPlanSaverLoader.java [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+                if(c.getCourseCode().equals(course2)){
+                    studyPlan.addCourseToSchedule(c, year, studyPeriod, 2);
+                    break;
+                }
+            }
+        }
+    }
+
+    @Override
+    public void createNewStudentFile(){
```
Comment (Eyzic) [inline src/main/java/ProjectOcean/IO/StudyPlanSaverLoader.java:238]: Missing javadoc

File: src/main/java/ProjectOcean/Main.java [hunk 2]
```diff
@@ -11,14 +11,22 @@
...
     @Override
     public void start(Stage primaryStage) {
         primaryStage.setTitle("Course Planning System");
-        ApplicationController root = new ApplicationController(getHostServices());
+        root = new ApplicationController(getHostServices());
         primaryStage.setScene(new Scene(root, 1200, 800));
         primaryStage.show();
+    }
 
+    @Override
```
Comment (Eyzic) [inline src/main/java/ProjectOcean/Main.java:26]: Feels stupid, but javadoc

File: src/main/java/ProjectOcean/Model/Course.java [hunk 3]
```diff
@@ -1,94 +1,130 @@
...
+     */
+    @Override
+    public String getLanguage() {
+        return cAccessibility.getLanguage();
     }
 
-    public String getCoursePMLink() {
-        return coursePMLink;
+    @Override
+    public boolean equals(Object o) {
```
Comment (Eyzic) [inline src/main/java/ProjectOcean/Model/Course.java:118]: Do you need javadoc on overidden methods? If yes then missing javadoc

File: src/main/java/ProjectOcean/Model/Course.java [hunk 4]
```diff
@@ -1,94 +1,130 @@
...
+        boolean b = studyPeriod.equals(course.studyPeriod) &&
+                cDetails.equals(course.cDetails) &&
+                cInfo.equals(course.cInfo) &&
+                cAccessibility.equals(course.cAccessibility);
+        return b;
     }
 
-    public String getCourseDescription() {
-        return courseDescription;
+    @Override
```
Comment (Eyzic) [inline src/main/java/ProjectOcean/Model/Course.java:132]: Do you need javadoc on overidden methods? If yes then missing javadoc

File: src/test/java/ProjectOcean/Model/CourseTests.java [hunk 5]
```diff
@@ -4,20 +4,175 @@
...
+        CourseFactory.SetCourseInfo("DAT017","Maskinorienterad programmering", "7.5");
+        CourseFactory.SetCourseDetails(new ArrayList<>(), "www.google.com", "Lorem Ipsum");
+        CourseFactory.SetCourseAccessibility("Rolf Söderström", "Tenta", "Svenska");
+        ICourse course2 = CourseFactory.CreateCourse();
+
+        //Checks that ICourse references are different
+        Assert.assertNotSame(course1, course2);
+    }
+
+    //These tests come from CPS test, but where removed because the methods where removed from CPS.
```
Comment (Eyzic) [inline src/test/java/ProjectOcean/Model/CourseTests.java:64]: A lot of outcommented code. Fix before merge with master.

File: src/test/java/ProjectOcean/Model/ScheduleTests.java [hunk 6]
```diff
@@ -1,4 +1,76 @@
...
+        Assert.assertTrue(schedule.getYear(2) != null);
+
+        schedule.removeYear(year);
+        Assert.assertTrue(schedule.getYear(2) == null);
+
+
+    }
+
+
+
```
Comment (Eyzic) [inline src/test/java/ProjectOcean/Model/ScheduleTests.java]: Make the format more clean. Kill empty lines.

File: src/test/java/ProjectOcean/Model/StudentTests.java [hunk 7]
```diff
@@ -1,4 +1,75 @@
...
+        CourseFactory.SetCourseAccessibility("Rolf Söderström", "Tenta", "Svenska");
+        Course course = CourseFactory.CreateCourse();
+        student.addCourse(course, year, studyPeriod, slot);
+
+        Assert.assertEquals(course, student.getCurrentStudyPlan().getSchedule().getYear(year).getStudyPeriod(studyPeriod).getCourse1());
+
+        student.removeCourse(year, studyPeriod, slot);
+        Assert.assertEquals(null, student.getCurrentStudyPlan().getSchedule().getYear(year).getStudyPeriod(studyPeriod).getCourse1());
+    }
+
```
Comment (Eyzic) [inline src/test/java/ProjectOcean/Model/StudentTests.java:50]: Delete a line

File: src/test/java/ProjectOcean/Model/StudyPeriodTests.java [hunk 8]
```diff
@@ -1,4 +1,77 @@
...
+
 public class StudyPeriodTests {
+
+    private int slot;
+
+    @Before
+    public void before() {
+        slot = 1;
+    }
+
```
Comment (Eyzic) [inline src/test/java/ProjectOcean/Model/StudyPeriodTests.java:19]: Delete one line

File: src/test/java/ProjectOcean/Model/StudyPlanTests.java [hunk 9]
```diff
@@ -1,4 +1,74 @@
...
+        CourseFactory.SetCourseInfo("DAT017","Maskinorienterad programmering", "7.5");
+        CourseFactory.SetCourseDetails(new ArrayList<>(), "www.google.com", "Lorem Ipsum");
+        CourseFactory.SetCourseAccessibility("Rolf Söderström", "Tenta", "Svenska");
+        Course course = CourseFactory.CreateCourse();
+
+        studyPlan.addCourseToSchedule(course, year, studyPeriod, slot);
+
+        Assert.assertEquals(course, studyPlan.getSchedule().getYear(year).getStudyPeriod(studyPeriod).getCourse1());
+
+
```
Comment (Eyzic) [inline src/test/java/ProjectOcean/Model/StudyPlanTests.java]: Delete empty lines

File: src/test/java/ProjectOcean/WorkspaceTest.java [hunk 10]
```diff
@@ -1,106 +1,170 @@
...
-        Workspace ws = new Workspace();
-        Course course = new Course("BAT123", "Beroendespecifika paradigmer", 7.5F,3,"Anders Bölinge", "Tenta","Turkish", null, "LINK_Zelda", "Abu Dhabi. YOLO!");
-        Course course2 = new Course("CAT123", "Complex system", 7.5F, 2, "Anders Fölinge", "Tenta","Turkish", null, "LINK_Zelda", "Abu Dhabi. YOLO!");
-        Course course3 = new Course("DAT321", "Datavetenskap", 7.5F, 4, "Anders Bölinge", "Tenta","Turkish", null, "LINK_Zelda", "Abu Dhabi. YOLO!");
-        Course course4 = new Course("FAT321", "Fysik för ingenjörer", 7.5F, 1, "Anders Brölinge", "Tenta","Turkish", null, "LINK_Zelda", "Abu Dhabi. YOLO!");
+    private Workspace ws = new Workspace();
+    private List courses = new LinkedList<Course>();
+
+    @Before
+    public void createCourses() {
```
Comment (Eyzic) [inline src/test/java/ProjectOcean/WorkspaceTest.java:22]: Change from LINK and Abu Dhabi to something more professional. Applies to all the tests below.

**Review by Eyzic — CHANGES_REQUESTED** [review]
Looks good overall. Look through the test files and format so that empty lines are gone and make sure it follows our code standards.  Some changes needed on some unprofessional strings.

File: src/test/java/ProjectOcean/Model/ScheduleTests.java (hunk 6 above)
Comment (Tetre2) [reply src/test/java/ProjectOcean/Model/ScheduleTests.java]: Dom är dödade!

**Review by Eyzic — APPROVED** [review]

**Pull request merged**